numpy/ma/tests/test_core.py

```diff
@@ -2126,7 +2126,11 @@ def test_assign_dtype(self):
         a = np.zeros(4, dtype='f4,i4')
 
         m = np.ma.array(a)
-        m.dtype = np.dtype('f4')
+        with warnings.catch_warnings():
```

Comment:
Here for sure use `pytest.warns` - we do want to check warnings are emitted also for `MaskedArray` (and also below of course)